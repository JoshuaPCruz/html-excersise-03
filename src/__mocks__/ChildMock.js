import React from 'react';

const ChildMock = () => {
  return (
    <>
      <h3>Título</h3>
      <p className="Content">Some text</p>
      <a href="" className="Button">Leer más</a>
    </>
  );
};

export default ChildMock;