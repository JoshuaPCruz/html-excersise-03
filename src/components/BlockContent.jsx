import React from 'react';
import '../css/components/BlockContent.css';
import '../css/components/Button.css';

const BlockContent = ({ children }) => {
  return (
    <div className="Block-Content">
      {children}
    </div>
  );
};

export default BlockContent;