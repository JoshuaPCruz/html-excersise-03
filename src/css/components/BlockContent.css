.Block-Content {
  max-width: 368px;
  min-width: 368px;
  max-height: 588px;
  min-height: 588px;
  position: relative;
}

.Block-Content h3 {
  margin-bottom: 19px;
  padding-bottom: 15px;
  border-bottom: 2px solid var(--gray-primary);
  color: var(--gray-secondary);
  font-size: 38px;
}

.Block-Content .Content {
  margin-bottom: 29px;
  font-size: 17px;
  line-height: 1.8em;
  color: var(--gray-primary);
}

.Block-Content .Content-Date {
  margin-bottom: 24px;
  font-size: 17px;
  line-height: 1.8em;
  color: var(--gray-secondary);
}

.Block-Content p span {
  color: var(--orange-secondary);
}

.Block-Content .Content-Date {
  margin-bottom: 24px;
  font-size: 17px;
  line-height: 1.8em;
  color: var(--gray-secondary);
}

.Block-Content ul {
  list-style: none;
}

.Block-Content ul li {
  margin-bottom: 14px;
  color: var(--orange-secondary);
  font-size: 17px;
  line-height: 1.8em;
}

.Block-Content ul li::first-letter {
  color: var(--gray-secondary);
}

.Block-Content .Button {
  position: absolute;
  bottom: 0;
  left: 0;
}