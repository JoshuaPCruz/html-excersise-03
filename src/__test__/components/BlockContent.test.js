import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import ChildMock from '../../__mocks__/ChildMock';
import BlockContent from '../../components/BlockContent';

describe('<BlockContent></BlockContent>', () => {
  const blockContent = shallow(
    <BlockContent>
      <ChildMock />
    </BlockContent>
  );
  test('Render del componente BlockContent', () => {
    expect(blockContent.length).toEqual(1);
  });

  test('Render del children', () => {
    expect(blockContent.contains(<ChildMock />)).toBeTruthy();
  });
});

describe('BlockContent Snapshot', () => {
  test('Comprobar la UI del componente BlockContent', () => {
    const blockContent = create(
      <BlockContent>
        <ChildMock />
      </BlockContent>
    );
    expect(blockContent.toJSON()).toMatchSnapshot();
  });
});