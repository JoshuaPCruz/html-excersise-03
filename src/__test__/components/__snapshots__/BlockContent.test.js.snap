// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockContent Snapshot Comprobar la UI del componente BlockContent 1`] = `
<div
  className="Block-Content"
>
  <h3>
    Título
  </h3>
  <p
    className="Content"
  >
    Some text
  </p>
  <a
    className="Button"
    href=""
  >
    Leer más
  </a>
</div>
`;
